src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +392,402 @@ else if (!period.isEmpty()) {
+                    break;
+                }
+            }
+        }
+
+        final DetailNode inlineHtmlNode = getInlineTagNodeInsideHtml(ast);
+        if (inlineHtmlNode != null) {
+            final DetailNode htmlDescriptionNode = getDescriptionNodeInsideHtml(inlineHtmlNode);
+            for (DetailNode child : htmlDescriptionNode.getChildren()) {
+                if (child.getType() == JavadocTokenTypes.TEXT) {
+                    result.append(child.getText());
+                }
+            }
+        }
+        return result.toString().trim();
+    }
+
+    /**
+     * Checks if the string ends with period.
+     *
+     * @param sentence string to check for period at end.
+     * @param period   String to check within sentence.
+     * @return true, is sentence ends with period.
+     */
+    private static boolean isPeriodAtEnd(String sentence, String period) {
+        final String summarySentence = sentence.trim();
+        return summarySentence.lastIndexOf(period) != summarySentence.length() - 1;
+    }
+
+    /**
+     * Checks if the summary is valid.
+     *
+     * @param ast Javadoc root node.
+     * @return true, if sentence is invalid.
+     */
+    private boolean isSummaryValid(DetailNode ast) {
+        final String inlineFirstSentence = getInlineSummary(ast);
+        final int endOfInlineSentence = inlineFirstSentence.lastIndexOf(period);
+        return containsForbiddenFragment(inlineFirstSentence.substring(0,
+                endOfInlineSentence))
+                || !containsCorrectInlineTags(ast)
+                || !validInlineTagInsideHtmlFormat(ast);
+    }
+
+    /**
+     * Finds and return if the summary sentence has allowed Inline Tags.
+     *
+     * @param ast Children of javadoc Inline Tag DetailNode[].
+     * @return true, if first sentence contains allowed tags.
+     */
+    private static boolean containsCorrectInlineTags(DetailNode ast) {
+        boolean contains = true;
+        for (DetailNode node : ast.getChildren()) {
+            final DetailNode[] child = node.getChildren();
+            if (child.length > 1
+                    && child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                    && getInlineTag(child) != null
+                    && !ALLOWED_INLINE_TAGS.contains(getInlineTag(child).getText())) {
+                contains = false;
+                break;
+            }
+        }
+        return contains;
+    }
+
+    /**
+     * Returns inline tag Node of html type summary.
+     *
+     * @param ast Root node.
+     * @return DetailNode of inline tag.
+     */
+    private static DetailNode getInlineTagNodeInsideHtml(DetailNode ast) {
+        DetailNode inlineNode = null;
+        for (DetailNode node : ast.getChildren()) {
+            for (DetailNode child : node.getChildren()) {
+                if (child.getType() == JavadocTokenTypes.PARAGRAPH) {
+                    inlineNode = getInlineTagNodeFromParagraph(child);
```

Comment:
this is not good. this check should not care about any HTML tags (like PARAGRAPH) is special way. There couold be any tags

Reply:
Done.